components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHttpBinding.java

```diff
@@ -319,7 +321,8 @@ public void populateCamelHeaders(ClientResponse response, Map<String, Object> he
 
     @Override
     public Object toHttpResponse(HttpServerExchange httpExchange, Message message) throws IOException {
-        boolean failed = message.getExchange().isFailed();
+        Exchange camelExchange = message.getExchange();
+        boolean failed = camelExchange.isFailed();
         int defaultCode = failed ? 500 : 200;
 
         int code = message.getHeader(Exchange.HTTP_RESPONSE_CODE, defaultCode, int.class);
```

Comment:
I was on thinking something simpler for the logic, maybe something like this:  
```suggestion
            int code = message.getHeader(Exchange.HTTP_RESPONSE_CODE, defaultCode, int.class);
            httpExchange.setStatusCode(code);
            Object body = message.getBody();
            code = checkIfNoContentAndReturnCorrectStatusCode(body);
            httpExchange.setStatusCode(code);
```
whereby `checkIfNoContentAndReturnCorrectStatusCode` will check if body is null as well if it has empty content, if true, then return `204`. As a result you won't need further else, What do you think of this approach ?

Reply:
oddly that was very close to what I originally had, though I would still prefer the method handled all of the changes inside of it since there are couple of headers and possibly the body itself

Reply:
also still have to handle 3 possible values for body differently
- null
- `InputStream`
- non `InputStream`